library/ssl_tls.c

```diff
@@ -1363,10 +1363,31 @@ int mbedtls_ssl_set_session( mbedtls_ssl_context *ssl, const mbedtls_ssl_session
         return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
     }
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3) && \
+    defined(MBEDTLS_SSL_SESSION_TICKETS)
+    if( ssl->handshake->resume == 1 )
+    {
+        if( session->tls_version == MBEDTLS_SSL_VERSION_TLS1_3 &&
+            ssl->session_negotiate->tls_version == MBEDTLS_SSL_VERSION_TLS1_3 )
+        {
+            mbedtls_ssl_session *next_session = ssl->session_negotiate;
+            target = mbedtls_calloc( 1, sizeof(mbedtls_ssl_session) );
+            if( target == NULL )
+                return( MBEDTLS_ERR_SSL_ALLOC_FAILED );
+
+            while( next_session->next != NULL )
+                next_session = next_session->next;
+            next_session->next = target;
+        }
+        else
+            return( MBEDTLS_ERR_SSL_FEATURE_UNAVAILABLE );
+    }
+#else /* MBEDTLS_SSL_PROTO_TLS1_3 && MBEDTLS_SSL_SESSION_TICKETS */
```

Comment:
I meant that eventually we should not change this function from what it is currently on development. Thus this change should be reverted and we should use `mbedtls_ssl_session_free_chain` when we build internally a chain of session object to free it.
```suggestion
```
```

Reply:
Do you mean move it to #6206 ?

Reply:
It is moved to #6206

Reply:
I do not see why we would need this in #6206 but we will see.

Reply:
Sorry I looked at this too quickly. I thought it was still the change in `mbedtls_ssl_session_free` but no it is the change in `mbedtls_ssl_set_session`. Please restore this thus remove the last commit and sorry for the noise.